### ISSUE TITLE
php mongo driver 1.6.2 and reference priming

### ISSUE THREAD

**Issue by vkartaviy** [doctrine/mongodb-odm#1030]
After update to the latest version ```1.6.2``` of ```mongo``` driver I began receiving empty result from queries that used ```prime``` feature. After some tests I found that it only happens when ```limit``` is used and limit value is <= then total number of documents in collection.

I have created [failing test case](https://github.com/vkartaviy/mongodb-odm/blob/mongodb-2.6.7/tests/Doctrine/ODM/MongoDB/Tests/Functional/ReferencePrimerTest.php) and the result you can see on [travis](https://travis-ci.org/vkartaviy/mongodb-odm/builds/50763635).

System info:
```
> uname -a
Linux xxx 3.2.0-4-amd64 #1 SMP Debian 3.2.65-1 x86_64 GNU/Linux

> php -v
PHP 5.5.21-1~dotdeb.1 (cli) (built: Feb  1 2015 19:51:51)
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2014 Zend Technologies
    with Zend OPcache v7.0.4-dev, Copyright (c) 1999-2014, by Zend Technologies

> mongod --version
db version v2.6.7
2015-02-14T06:02:00.274+0000 git version: a7d57ad27c382de82e9cb93bf983a80fd9ac9899

> php --ri mongo | grep Version
Version => 1.6.2
```

**Comment by vkartaviy** [comment]
@jmikola @derickr It looks like that is ```MongoCursor``` related issue. Please have a look.

**Comment by derickr** [comment]
Yeah, there seems to be a bug here.

**Comment by derickr** [comment]
@vkartaviy I have filed it as https://jira.mongodb.org/browse/PHP-1394 and there is a PR up at https://github.com/mongodb/mongo-php-driver/pull/827 — if you could test the https://github.com/derickr/mongo-php-driver/tree/issue1394-getnext-with-limit branch, that'd be awesome.

**Comment by vkartaviy** [comment]
@derickr Nope. It still fails. It looks like cursor is dies after iterating through all of the documents in it (for ex. ```iterator_to_array```) and it's not possible to reuse it anymore. Is it should be dead? If yes then bug in reference priming code in ```mongo-odm``` because it reuse the cursor iterator.

**Comment by vkartaviy** [comment]
@derickr That is the failing test for iterator reuse issue.

```
--TEST--
Test for PHP-1394: cursor reuse (with limit)
--SKIPIF--
<?php require_once "tests/utils/standalone.inc" ?>
--FILE--
<?php
require_once "tests/utils/server.inc";

function log_getmore($server, $info) {
        echo "Issuing getmore\n";
}

$ctx = stream_context_create(array(
        'mongodb' => array(
                'log_getmore' => 'log_getmore',
        ),
));

$host = MongoShellServer::getStandaloneInfo();
$m = new MongoClient($host, array(), array('context' => $ctx));

$c = $m->selectCollection(dbname(), collname(__FILE__));
$c->drop();

$c->save(array('_id' => 'test1'));
$c->save(array('_id' => 'test2'));
$c->save(array('_id' => 'test3'));
$c->save(array('_id' => 'test4'));
$cur = $c->find(array(), array('_id'))->limit(2);
var_dump(count(iterator_to_array($cur)));
var_dump(count(iterator_to_array($cur)));

?>
--EXPECTF--
int(2)
int(2)
```

**Comment by derickr** [comment]
Ok, @vkartaviy - that seems to have been a different issue. PHP-1394 is merged into v1.6 now, and I have created https://jira.mongodb.org/browse/PHP-1395 for the second issue. PR for that is ready for review too (and is based on PHP-1394, so feel free to test that).

**Comment by vkartaviy** [comment]
@derickr Well done! Now it works as [expected](https://travis-ci.org/vkartaviy/mongodb-odm/builds/51099755). :+1:
